src/aleph/netty.clj

```diff
@@ -203,41 +200,48 @@
   (->> (bs/convert x (bs/stream-of ByteBuf) {:chunk-size chunk-size})
     (s/onto nil)))
 
+(defn ensure-dynamic-classloader
+  "Ensure the context class loader has a valid loader chain to
+   prevent `ClassNotFoundException`.
+   https://github.com/clj-commons/aleph/issues/603."
+  []
+  (let [thread (Thread/currentThread)
+        context-class-loader (.getContextClassLoader thread)
+        compiler-class-loader (.getClassLoader clojure.lang.Compiler)]
+    (when-not (instance? DynamicClassLoader context-class-loader)
+      (.setContextClassLoader
+        thread
+        (DynamicClassLoader. (or context-class-loader
+                                 compiler-class-loader))))))
+
+(defn- operation-complete [^Future f d]
+  (cond
+     (.isSuccess f)
+     (d/success! d (.getNow f))
+
+     (.isCancelled f)
+     (d/error! d (CancellationException. "future is cancelled."))
+
+     (some? (.cause f))
+     (if (instance? java.nio.channels.ClosedChannelException (.cause f))
+       (d/success! d false)
+       (d/error! d (.cause f)))
+
+     :else
+     (d/error! d (IllegalStateException. "future in unknown state"))))
+
 (defn wrap-future
   [^Future f]
   (when f
     (if (.isSuccess f)
       (d/success-deferred (.getNow f) nil)
       (let [d (d/deferred nil)
-            ;; workaround for the issue with executing RT.readString
-            ;; on a Netty thread that doesn't have appropriate class loader
-            ;; more information here:
-            ;; https://github.com/clj-commons/aleph/issues/365
-            class-loader (or (clojure.lang.RT/baseLoader)
-                             (clojure.lang.RT/makeClassLoader))]
+            bound-operation-complete (bound-fn* operation-complete)]
```

Comment:
We should keep a comment explaining it all. Since it's on a netty thread, the vars and classloader will be different

Reply:
95b67d4a66c100c7b045fc4553241c3bf7a38dbf

Can I ask you if this comment is enough?